experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller/keygen.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```

Comment:
It is a good practice to have "fail fast" bash error flagging. Your bash will stop executing as soon as it encountered an error.

I generally use `set -e` . You can improve the flags so that it fails cleanly when using piped commands. But this is not required in this bash

Reply:
`set -eo pipefail` ?